Automattic/wp-calypso
# Jetpack Connect Authorize: XMLRPC error fallback

What's going on:
Currently in Jetpack, the connection process happens in wp-admin. Here in calypso we are attempting to do it via an XMLRPC request. Some hosts limit XMLRP requests, which will result in more failed connections using the calypso flow.

This PR attempts to detect XMLRPC errors and fallback to the flow within wp-admin.

To test:
- get this branch going at calypso.localhost:3000
- go to wp-admin dashboard of one of your unconnected Jetpack sites
- click 'connect'
- on the next screen, add `&jetpack_connect_test=calypso&calypso_env=development` to the (really long) URL in your address bar, and press enter
- disable XMLRPC on your Jetpack site, ie rename xmlrpc.php to something else
- click 'Approve' on the calypso screen
- ensure that you receive this improved error screen

<img width="464" alt="screen shot 2016-08-22 at 4 34 22 pm" src="https://cloud.githubusercontent.com/assets/2694219/17870726/94a13fea-6886-11e6-878a-25626762b518.png">


cc: @kraftbj @johnHackworth @tyxla 

Test live: https://calypso.live/?branch=update/jetpack-connect-xmlrpc-error-fallback
